modules/auxiliary/gather/ldap_query.rb

```diff
@@ -263,6 +269,36 @@ def output_data_csv(entries)
     generate_rex_tables(entries, 'csv')
   end
 
+  def find_root_dn(ldap)
+    filter = '(objectClass=*)'
+    attributes = ['instanceType']
+    @root_dn = @base_dn
+    loop do
+      attributes_data = perform_ldap_query(ldap, filter, attributes, base: @root_dn)
+      for entry in attributes_data do
+        int_instance_type = entry[:instancetype][0].to_i
+        # Check if this is a root of a naming context (NC).
+        if (int_instance_type & IT_NC_HEAD)
```

Comment:
Is this valid? I believe `0` is truthy in Ruby, so this will always enter this if statement

Or is `&` overridden to return true/false ?

Reply:
Similar to the 'never terminating' logic comment above; If `if (int_instance_type & IT_NC_HEAD) == 1` evaluated to to false - would this just be an infinite loop if it never found found the base case? 🤔

Reply:
I thought this was a bitwise AND operation with the value of `IT_NC_HEAD` aka 1?

Reply:
I believe 0 is truthy in Ruby, so this will always enter this if statement

```ruby
if 0
  puts '0 is truthy'
end

if 1
  puts '1 is also truthy'
end
```

Result:

```
$ bundle exec ruby main.rb
0 is truthy
1 is also truthy
```

Reply:
Oh nm I see the issue now:

```
irb(main):009:1* if(9 & 1)
irb(main):010:1*   print("A")
irb(main):011:0> end
A=> nil
irb(main):012:1* if(8 & 1)
irb(main):013:1*   print("A")
irb(main):014:0> end
A=> nil
irb(main):015:0> 
```

Reply:
So most likely missing the `==` operator to convert the bit mask integer result into a boolean

```
if (int_instance_type & IT_NC_HEAD) == 1
```

Reply:
Removed this and loop and the constants with the revised update.